archunit/src/test/java/com/tngtech/archunit/core/domain/JavaMethodTest.java

```diff
@@ -0,0 +1,58 @@
+package com.tngtech.archunit.core.domain;
+
+
+import com.tngtech.archunit.core.importer.ClassFileImporter;
+import org.junit.Test;
+
+import static com.tngtech.archunit.testutil.Assertions.assertThat;
+
+public class JavaMethodTest {
+    @Test
+    public void isOverriddenTest() {
+        class Base {
+            void method1() {
+            }
+
+            void method1(int x) {
+            }
+        }
+        class Child extends Base {
+            void method1() {
+            }
+
+            void method2() {
+            }
+        }
+        class GrandChild extends Child {
+            void method1() {
+
+            }
+
+            void method1(int x) {
+
+            }
+
+            void method2() {
+
+            }
+
+            void method3() {
+
+            }
+
+        }
+        ClassFileImporter importer = new ClassFileImporter();
+        JavaClass baseClass = importer.importClass(Base.class);
+        JavaClass childClass = importer.importClass(Child.class);
+        JavaClass grandChildClass = importer.importClass(GrandChild.class);
+        assertThat(baseClass.getMethod("method1").isOverridden()).isFalse();
+        assertThat(baseClass.getMethod("method1", int.class).isOverridden()).isFalse();
+        assertThat(childClass.getMethod("method1").isOverridden()).isTrue();
+        assertThat(childClass.getMethod("method2").isOverridden()).isFalse();
+        assertThat(grandChildClass.getMethod("method1").isOverridden()).isTrue();
+        assertThat(grandChildClass.getMethod("method1", int.class).isOverridden()).isTrue();
+        assertThat(grandChildClass.getMethod("method2").isOverridden()).isTrue();
+        assertThat(grandChildClass.getMethod("method3").isOverridden()).isFalse();
+        //TODO add testing for methods with generic parameters
+    }
+}
```

Comment:
We should also have a test for the use case of #1047 .

Reply:
I believe that is already in place, currently we check if the parameters and name of the method are equal, and that implies compatibility of return type.
If that is not the case, then the user code will not compile in the first place

Reply:
I thought that we should document that this works (which it curently does, but we also want to avoid any future regression):
```java
    @Test
    public void more_specific_return_type_is_supported() {
        class Parent {
            CharSequence method() { return "base"; }
        }
        class Child extends Parent {
            @Override
            String method() { return "overridden"; }
        }
        JavaClass childClass = new ClassFileImporter().importClass(Child.class);
        JavaMethod method = childClass.getMethod("method");
        assertThat(method.isOverridden()).isTrue();
    }
```